src/role-access/containers/specific-access/specific-access-home/specific-access-home.component.ts

```diff
@@ -0,0 +1,125 @@
+  public specificAccessApprovedVisibilityStates = specificAccessApprovedVisibilityStates;
+  public specificAccessInformationVisibilityStates = specificAccessInformationVisibilityStates;
+  public specificAccessDeniedVisibilityStates = specificAccessDeniedVisibilityStates;
+
+  constructor(
+    private readonly store: Store<fromFeature.State>,
+    private readonly route: ActivatedRoute,
+    private readonly router: Router) {
+    this.caseId = this.route.snapshot.queryParams && this.route.snapshot.queryParams.caseId ?
+      this.route.snapshot.queryParams.caseId : '';
+  }
+
+  public ngOnInit(): void {
+    this.specificAccessStateDataSub = this.store.pipe(select(fromFeature.getSpecificAccessState)).subscribe(
+      specificAccessReviewStateData => {
+        this.navigationCurrentState = specificAccessReviewStateData.state;
+      }
+    );
+  }
+
+  public onNavEvent(event: SpecificAccessNavigationEvent): void {
+    this.navEvent = {
+      event,
+      timestamp: Date.now()
+    };
+    this.navigationHandler(event);
+  }
+
+  public isComponentVisible(currentNavigationState: SpecificAccessState, requiredNavigationState: SpecificAccessState[]): boolean {
+    return requiredNavigationState.includes(currentNavigationState);
+  }
+
+  public navigationHandler(navEvent: SpecificAccessNavigationEvent): void {
+    switch (navEvent) {
+      case SpecificAccessNavigationEvent.BACK: {
+        switch (this.navigationCurrentState) {
+          case SpecificAccessState.SPECIFIC_ACCESS_DURATION:
+            this.store.dispatch(new fromFeature.ChangeSpecificAccessNavigation(SpecificAccessState.SPECIFIC_ACCESS_REVIEW));
+            break;
+          default:
+            throw new Error('Invalid specific access state');
+        }
+        break;
+      }
+      case SpecificAccessNavigationEvent.CONTINUE: {
+        switch (this.navigationCurrentState) {
+          case SpecificAccessState.SPECIFIC_ACCESS_REVIEW:
+            this.specificAccessReviewComponent.navigationHandler(navEvent);
+            break;
+          case SpecificAccessState.SPECIFIC_ACCESS_DURATION:
+            this.specificAccessDurationComponent.navigationHandler(navEvent);
+            break;
+          case SpecificAccessState.SPECIFIC_ACCESS_APPROVED:
+            this.specificAccessDurationComponent.navigationHandler(navEvent);
+            break;
+          default:
+          // throw new Error('Invalid specific access state');
+        }
+        break;
+      }
+      case SpecificAccessNavigationEvent.RETURNTOMYTASKS: {
+        this.router.navigateByUrl(`/work/list`);
+        break;
+      }
+      case SpecificAccessNavigationEvent.RETURNTOTASKSTAB: {
+        this.router.navigateByUrl(`/cases/case-details/${this.caseId}/roles-and-access`);
+        break;
+      }
+      case SpecificAccessNavigationEvent.CANCEL:
+        //        this.router.navigateByUrl(`cases/case-details/${this.caseId}/roles-and-access`);
+        break;
```

Comment:
Cancel case could be removed if not needed anymore, or is this just for reference when things get wired up more in other tickets?

Reply:
Leaving in place for when things get wired up